js/Source/Bootstrap.Popup/Delegator.BS.ShowPopup.js

```diff
@@ -19,14 +19,24 @@ provides: [Delegator.BS.ShowPopup]
 ...
 */
 
-Delegator.register('click', 'BS.showPopup', {
+(function(){
 
-  handler: function(event, link, api){
-    var target = api.get('target') ? link.getElement(api.get('target')) : document.id(link.get('href').split("#")[1]);
-    event.preventDefault();
-    if (!target) api.fail('Could not find target element to activate: ' + (api.get('target') || link.get('href')));
-    api.getBehavior().apply(target);
-    target.getBehaviorResult('BS.Popup').show();
-  }
+  var triggers = {};
 
-});
\ No newline at end of file
+  ['show', 'hide'].each(function(action){
+
+    triggers['BS.'+ action + 'Popup'] = {
+      handler: function(event, link, api){
+        var target = api.get('target') ? link.getElement(api.get('target')) : document.id(link.get('href').split("#")[1]);
+        event.preventDefault();
+        if (!target) api.fail('Could not find target element to activate: ' + (api.get('target') || link.get('href')));
+        api.getBehavior().apply(target);
+        target.getBehaviorResult('BS.Popup')[action]();
```

Comment:
Good opportunity to update this. `api.getElement` does this but throws an error if it's not found.

This method was just written before I added `api.getElement`. So I'd change this to:

```
event.preventDefault();
var target = api.get('target') ? api.getElement('target') : document.id(...);
api.getBehavior().apply(target);
etc...
```

See `Delegator.addRemoveClass` for a working example. No material change; just gets rid of code.